packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -38,7 +39,10 @@ const store = import.meta.client && cookieStore ? window.cookieStore : undefined
 export function useCookie<T = string | null | undefined> (name: string, _opts?: CookieOptions<T> & { readonly?: false }): CookieRef<T>
 export function useCookie<T = string | null | undefined> (name: string, _opts: CookieOptions<T> & { readonly: true }): Readonly<CookieRef<T>>
 export function useCookie<T = string | null | undefined> (name: string, _opts?: CookieOptions<T>): CookieRef<T> {
-  const opts = { ...CookieDefaults, ..._opts }
+  const filter = (key: string) => {
+    return key !== name
+  }
+  const opts = { ...CookieDefaults, ..._opts, filter }
```

Comment:
I think we should only add filter if user hasn't provided their own.

Reply:
yes, you are right